### ISSUE TITLE
Pinterest-like modal dialog behavior

### ISSUE THREAD

**Issue by MrOrz** [christopherthielen/ui-router-extras#11]
TL;DR -- The view of sticky state is cleaned up after `$state.go()` calls.

---

It's really exciting to see the discussion in https://github.com/angular-ui/ui-router/issues/894 become modularized and usable. It's like a dream coming into reality. Thank you so much for this @christopherthielen !

The first thing I tried to build with ui-router-extras is a modal example. It slightly differs from the one in [your example](http://christopherthielen.github.io/ui-router-extras/example/stickymodal/index.html) because I want to support navigation between modal views without dismissing & re-initializing the existing modal dialog, as well as supporting the "back" button in the browser.

Using `$stateChangeStart` instead of `onEnter` & `onExit` callbacks, it can be easily achieved:
http://plnkr.co/edit/4KqUNnRaUDNZO9t3VD0e

---

As a follow-up, I wanted to implement pinterest-like modal window using ui-router-extras. When the URL of individual items is visited directly, it should be presented in full-size, instead of being confined inside a modal.

However, this time the try is not successful.

I followed @ashaffer 's solution in [the comment of the previous thread](https://github.com/angular-ui/ui-router/issues/894#issuecomment-36778322), resulting in the followig plunker:

http://plnkr.co/edit/8r7haRWCcpmT2ZmnSQCE

The core of the technique is to nest the "modal" state and the "full-size" state under a common parent state. In the parent state we redirect the user to either "modal" or "full-size" state according to where the user is from.

The full-sized view is working perfectly, as shown in the following link. When navigating to other items from the full-sized view, they were also shown in full-sized view, which is identical to the behavior of pinterest.

http://run.plnkr.co/plunks/8r7haRWCcpmT2ZmnSQCE/#/item/5

However, when a modal dialog is opened, the view of the sticky list state beneath is cleaned up, which can be reproduced by clicking any item in the following link:

http://run.plnkr.co/plunks/8r7haRWCcpmT2ZmnSQCE/#/

It seems that the effect of `sticky` goes away after the state redirect. Is this an intended behavior?

**Comment by TD-Stanley** [comment]
does anyone have an example / plunker of this functioning as properly?

**Comment by TD-Stanley** [comment]
Extremely sorry about pulling this back up from the dead:
I used your example in the plunker to get me in the direction I need., but how would I pass data back from the modal to the original state / controller? I've attempted using a service, and watching from the first controller, but the $watch doesn't work once the modal in invoked.  I've also tried passing it back in fromsStateParams. 

Given this use case, how would I post data back to the original controller from the modal?

**Comment by random-one** [comment]
@TD-Stanley I hope you're not using exactly this version of ui-bootstrap, as the one in the plunkr. The module has more recent versions. The cause of your problem is that you're not using all the features `ui-bootstrap` offers. Please take a look at the documentation for version 0.13.1 which I saw is the first that offers a solution.
To be more detailed there is a property called `bindToController` that shares the scope of the controller that opens the $modal (and it's called $uibModal in newer versions) with the $uibModal instance.

**Comment by TD-Stanley** [comment]
Thanks @random-one random-one. Seems like that should work. I'll give it a shot.  kicking myself for not seeing that in the docs before my other unsuccessful attempts.

**Comment by TD-Stanley** [comment]
```
			// Open up modal
			modalInstance = $uibModal.open({
				template: '<div ui-view="modal"></div>',
        controller: 'post-modalCtrl',
        controllerAs: '$scope',
        bindToController: true,
        	resolve: {
            resolvedData: function (Post) {
            if(toStateParams.post !== null){
              return {'post': toStateParams.post, 'index': toStateParams.index};
            }else{
  					return Post.getPost({_id: toStateParams.postId}).then(function(res){
              return {'post': res.data[0], 'index': toStateParams.index};
            })
  				}
  			}
      }
    });

i'm trying this, but still no luck. Perhaps because the modal has its own state and is opened within the "modalState machine" conditional and not within the initial controller..
